### ISSUE TITLE
Reduce resource fragment

### ISSUE THREAD

**Issue by mzmssg** [microsoft/pai#2538]
**What would you like to be added**:
Reduce resource fragment in sheduler

**Why is this needed**:

To increase utilization

**Without this feature, how does the current module work**:

Pack job to the highest utilized node.

**Components that may involve changes**:
Launcher, yarn

**Design**
We will make the fragment job low priority, preemptible for large job, then large job won't be blocked by fragment. 
It is achieved by node-label mechinism, every job could provide a node label representing prefrred node, we will schedule guarantee resource on these nodes, or roll back to other nodes with opportunistic resource. Then admin could reserve some nodes for large job. 
New request fields:
1.	“node-label”, by default it’s VC_NAME:GPU_NUM, noticed that it's a label matained by launcher, not yarn node-label
2.	“bonus-enabled”, by default it’s false. If true, launcher will try to submit opportunistic request after guaranteed request failure, otherwise job will fail immediately.

New launcher request logic:
 
![image](https://user-images.githubusercontent.com/11887940/57204191-b3bb4080-6fe8-11e9-91ef-c5f17cf95b1a.png)

---------------------------
### Test Cases
Assume 2 4-gpu nodes cluster:
1. submit 1-GPU job Job-A and 2-GPU job Job-B, they should both on Node-A.
2. submit 2-GPU job Job-C, because Node-A have only 1 free GPU, Job-C should be on Node-B.
3. finish Job-B, then Node-A have 3 free GPU, Node-B have 2 free GPU, we have total 5 free GPU, but can't run a 4-GPU job.

**Comment by mzmssg** [comment]
#2013

**Comment by scarlett2018** [comment]
@mzmssg - May we design few test scenarios and set a baseline to compare "before" vs. "after"? thanks.

**Comment by mzmssg** [comment]
@scarlett2018 
Yeah, it will be added after finding current cause.

**Comment by scarlett2018** [comment]
Adding @squirrelsc , he had few feedbacks from customer that can input as our "Test Case". We need to make sure those test cases are addressed by the design and implementaiton. I'm adding a **Test Case** Setion in the first comment of the Issue, please directly update that field.

**Comment by squirrelsc** [comment]
One case is found by our customer, they have two machines with 8 CPU cores, and starts two jobs which requests 8 cores. The jobs show as running more than 1 day and not complete. It's highly suspicious that two AMs of jobs are assigned to two machines, so that there are only 7 cores left on each machine.

**Comment by mzmssg** [comment]
AM fragment is possible, because we don't take AM into consideration. It would be better with some job container case

**Comment by fanyangCS** [comment]
address this in pure k8s version.

**Comment by fanyangCS** [comment]
https://github.com/microsoft/pai/issues/2585

**Comment by fanyangCS** [comment]
we rely on HiveD for this.
